src/main/java/com/denizenscript/denizencore/objects/core/ImageTag.java

```diff
@@ -0,0 +1,386 @@
+        // Rescales an image.
+        // The input is a <@link ObjectType MapTag> with "width" and "height" keys.
+        // Both are optional, and default to the image's current respective value.
+        // @example
+        // Rescales an image to be 50x50
+        // - adjust def:image scale:[width=50;height=50]
+        // @example
+        // Makes an image taller, keeping its existing width.
+        // - adjust def:short_image scale:[height=100]
+        // @tags
+        // <ImageTag.width>
+        // <ImageTag.height>
+        // -->
+        tagProcessor.registerMechanism("scale", false, MapTag.class, (object, mechanism, input) -> {
+            ElementTag widthElement = input.getElement("width");
+            ElementTag heightElement = input.getElement("height");
+            if ((widthElement != null && !widthElement.isInt()) || (heightElement != null && !heightElement.isInt())) {
+                mechanism.echoError("Invalid width/height '" + widthElement  + '/' + heightElement + "' specified: must be valid numbers.");
+                return;
+            }
+            int width = widthElement != null ? widthElement.asInt() : object.image.getWidth();
+            int height = heightElement != null ? heightElement.asInt() : object.image.getHeight();
+            BufferedImage rescaled = new BufferedImage(width, height, object.image.getType());
+            Graphics2D graphics = rescaled.createGraphics();
+            graphics.setComposite(AlphaComposite.Src);
+            graphics.drawImage(object.image, 0, 0, width, height, null);
+            graphics.dispose();
+            object.image = rescaled;
+            object.markChanged();
+        });
+    }
+
+    public static final ObjectTagProcessor<ImageTag> tagProcessor = new ObjectTagProcessor<>();
+
+    @Override
+    public ObjectTag getObjectAttribute(Attribute attribute) {
+        return tagProcessor.getObjectAttribute(this, attribute);
+    }
+
+    @Override
+    public void adjust(Mechanism mechanism) {
+        tagProcessor.processMechanism(this, mechanism);
+    }
+
+    @Override
+    public void applyProperty(Mechanism mechanism) {
+        mechanism.echoError("Cannot apply properties to an ImageTag.");
+    }
+
+    public byte[] getRawBytes() {
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        try {
+            ImageIO.write(image, imageType, outputStream);
+            return outputStream.toByteArray();
+        }
+        catch (IOException e) {
+            Debug.echoError("Exception when converting image to bytes:");
+            throw new RuntimeException(e);
+        }
+    }
+
+    String hexEncodedBytesCache;
+
+    public void markChanged() {
+        hexEncodedBytesCache = null;
+    }
+
+    @Override
+    public String identify() {
+        if (hexEncodedBytesCache == null) {
+            hexEncodedBytesCache = BinaryTag.hexEncode(getRawBytes(), false);
+        }
+        return "image@" + hexEncodedBytesCache;
```

Comment:
Probably should store `image@` in the cache so it's not copying huge strings in memory here